src/json_schema/parsing.rs

```diff
@@ -239,6 +239,10 @@ pub fn parse_ref(
         .as_str()
         .ok_or_else(|| anyhow!("'$ref' must be a string"))?;
 
+    if ref_path == "#" {
+        return Err(anyhow!("Recursive references are not supported for now"));
+    }
```

Comment:
Does this cover all cases of recursion (e.g. when the ref. path is not just `#`)?

Reply:
No, you're right, this is just the most obvious one, which doesn't take away support of internal referencing, which still could be used to create a recursion, like:

```
        {
            "type": "object",
            "properties": {
                "node": { "$ref": "#/definitions/node" }
            },
            "definitions": {
                "node": {
                    "type": "object",
                    "properties": {
                        "value": { "type": "integer" },
                        "next": { "$ref": "#/definitions/node" }
                    }
                }
            }
        }
```

I see these ways to proceed here:
1) this partial fix
2) turning off all referencing completely (well internal/local, external isn't supported right now), until we implement a recursive support as a feature

Because in order to keep internal referencing we'd need to do a full walk through to detect recursion, which is 99% of the recursive feature, I suppose.

Reply:
> Because in order to keep internal referencing we'd need to do a full walk through to detect recursion, which is 99% of the recursive feature, I suppose.

This is the only approach that will actually address the problem, and I think we should add the recursive feature, so this would be my recommendation.

@rlouf?

Reply:
Let's fully address the problem, I don't see any other viable choice here.